test/suite/test_rollback_to_stable33.py

```diff
@@ -0,0 +1,195 @@
+
+# test_rollback_to_stableXX.py
+# Test interaction between fast-delete and RTS.
+class test_rollback_to_stableXX(test_rollback_to_stable_base):
+    session_config = 'isolation=snapshot'
+    conn_config = 'cache_size=50MB,statistics=(all),log=(enabled=false)'
+
+    format_values = [
+        ('column', dict(key_format='r', value_format='S', extraconfig='')),
+        ('column_fix', dict(key_format='r', value_format='8t',
+            extraconfig=',allocation_size=512,leaf_page_max=512')),
+        ('integer_row', dict(key_format='i', value_format='S', extraconfig='')),
+        ('string_row', dict(key_format='S', value_format='S', extraconfig='')),
+    ]
+    prepare_values = [
+        ('no_prepare', dict(prepare=False)),
+        ('prepare', dict(prepare=True)),
+    ]
+    second_checkpoint_values = [
+        ('second_checkpoint', dict(second_checkpoint=True)),
+        ('no_second_checkpoint', dict(second_checkpoint=False)),
+    ]
+    rollback_modes = [
+        ('runtime', dict(crash=False)),
+        ('recovery', dict(crash=True)),
+    ]
+
+    scenarios = make_scenarios(format_values, prepare_values, second_checkpoint_values,
+        rollback_modes)
+
+    # Make all the values different so it's easier to see what happens if ranges go missing.
+    def mkdata(self, basevalue, i):
+        if self.value_format == '8t':
+            return basevalue
+        return basevalue + str(i)
+
+    def evict(self, ds, lo, hi, basevalue):
+        evict_cursor = self.session.open_cursor(ds.uri, None, "debug=(release_evict)")
+        self.session.begin_transaction()
+
+        # Evict every 3rd key to make sure we get all the pages but not write them out
+        # over and over again any more than necessary. FUTURE: improve this to evict
+        # each page once when we get a suitable interface for that.
+        for i in range(lo, hi, 3):
+            evict_cursor.set_key(ds.key(i))
+            self.assertEquals(evict_cursor.search(), 0)
+            self.assertEquals(evict_cursor.get_value(), self.mkdata(basevalue, i))
+            evict_cursor.reset()
+        self.session.rollback_transaction()
+        evict_cursor.close()
+
+    # Call this checkx to distinguish it from the parent class's default check().
+    def checkx(self, ds, nrows, read_ts, basevalue):
+        self.session.begin_transaction('read_timestamp=' + self.timestamp_str(read_ts))
+        cursor = self.session.open_cursor(ds.uri)
+        i = 1
+        for k, v in cursor:
+            self.assertEqual(v, self.mkdata(basevalue, i))
+            self.assertEqual(k, ds.key(i))
+            i += 1
+        self.session.commit_transaction()
+        self.assertEqual(i, nrows + 1)
+        cursor.close()
+
+    def test_rollback_to_stable(self):
+        # RTS will fail if there are uncommitted prepared transactions, so skip tests of prepare
+        # with a runtime call to RTS, that doesn't add useful testing scenarios.
+        if self.prepare and not self.crash:
```

Comment:
The test still fails on prepared?

Reply:
RTS will fail any time there are active transactions, won't it?

Reply:
Oh, that's correct.